doofinder/includes/class-update-on-save.php

```diff
@@ -0,0 +1,224 @@
+        $table_exists = $wpdb->get_var("SHOW TABLES LIKE '" . $table_name . "'");
+
+        if ($table_exists === NULL) {
+            // The table does not exist, we create it
+            $charset_collate = $wpdb->get_charset_collate();
+            $sql = "CREATE TABLE $table_name (
+                post_id INT NOT NULL,
+                type_post VARCHAR(255),
+                type_action VARCHAR(255),
+                PRIMARY KEY (post_id)
+            ) $charset_collate;";
+            require_once(ABSPATH . 'wp-admin/includes/upgrade.php');
+            dbDelta($sql);
+        }
+
+        update_option( 'doofinder_update_on_save_last_exec', date('Y-m-d H:i:s') );
+	}
+
+    /**
+     * Register hooks and actions.
+     */
+    public static function register_hooks() {
+        add_action('wp_insert_post', function ($post_id, \WP_Post $post, $updated) {
+            self::add_item_to_update($post_id, $post, $updated);
+        }, 99, 3);
+    }
+
+    /**
+     * Determines if the current post should be indexed and adds it to the update queue.
+     *
+     * @param int $post_id The ID of the post.
+     * @param WP_Post $post The post object.
+     * @param bool $updated Whether the post has been updated.
+     */
+	public static function add_item_to_update($post_id, $post, $updated) {
+
+        $log = new Log( 'update_on_save.txt' );
+        $update_on_save_index = new Update_On_Save_Index();
+
+        if (Settings::is_update_on_save_enabled()) {
+            $doofinder_post = new Post($post);
+            $log->log( 'Add this item to update on save: ' . print_r($doofinder_post, true)); 
+
+            self::add_item_to_db($doofinder_post, $post_id, $post->post_status, $post->post_type);
+
+            if(self::allow_process_items()) {
+                $log->log( 'We can send the data. '); 
+                $update_on_save_index->launch_doofinder_update_on_save();
+                return self::clean_update_on_save_db();
+            }
+        }
+
+        return;
+	}
+
+    /**
+     * Adds the item to the update queue in the database.
+     *
+     * @param Post $doofinder_post The Doofinder post object.
+     * @param int $post_id The ID of the post.
+     * @param string $status The status of the post.
+     * @param string $type The type of the post.
+     */
+    public static function add_item_to_db($doofinder_post, $post_id, $status, $type) {
+        $log = new Log( 'update_on_save.txt' );
+        if ( $status === 'auto-draft' || $type === "revision") {
+            $log->log( 'It is not necessary to save it as it is a draft. '); 
+        } elseif ( $doofinder_post->is_indexable()) {
+            $log->log( 'The item will be saved with the update action. '); 
+            self::add_to_update_on_save_db($post_id, $type, "update");
+        } else {
+            $log->log( 'The item will be saved with the delete action. '); 
+            self::add_to_update_on_save_db($post_id, $type, "delete");
+        }
+        return;
```

Comment:
I don't get to understand this logic. Also, I don't think the return is necessary here.

Reply:
This logic is simple.
- If it is draft or revision we don't need to do anything with it because we don't have to index it.
- If the status of the post is still indexable after the changes we do an update.
- If the status of the post is no longer indexable we have to delete it.

I am going to put a comment so that it is better understood. But you are right, the bad habit of the empty return. @Aitor-Corrales